From 08c4167790785c112357fa769b3e0f11654abd2b Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 6 Mar 2024 19:17:15 +0100
Subject: [PATCH 18/59] FIPS: skip fipsinstall tests

Patch-name: 0034.fipsinstall_disable.patch
Patch-id: 34
Patch-status: |
    # # Skip fipsinstall tests in the Red Hat FIPS build
From-dist-git-commit: 4334bc837fbc64d14890fdc51679a80770d498ce
---
 test/recipes/00-prep_fipsmodule_cnf.t |  10 ++++++----
 test/recipes/01-test_fipsmodule_cnf.t |   7 ++++---
 test/recipes/03-test_fipsinstall.t    |   2 +
 3 files changed, 12 insertions(+), 7 deletions(-)
 mode change 100644 => 100755 test/recipes/00-prep_fipsmodule_cnf.t
 mode change 100644 => 100755 test/recipes/01-test_fipsmodule_cnf.t
 mode change 100644 => 100755 test/recipes/03-test_fipsinstall.t

diff --git a/test/recipes/00-prep_fipsmodule_cnf.t b/test/recipes/00-prep_fipsmodule_cnf.t
old mode 100644
new mode 100755
index 4e3a6d85e8..48869b2568
--- a/test/recipes/00-prep_fipsmodule_cnf.t
+++ b/test/recipes/00-prep_fipsmodule_cnf.t
@@ -29,8 +29,10 @@ my $fipsmoduleconf = bldtop_file('test', 'fipsmodule.cnf');
 
 plan tests => 1;
 
+ok(1, "fipsinstall setup skipped without FIPS provider module");
+
 # Create the $fipsmoduleconf file
-ok(run(app(['openssl', 'fipsinstall', '-pedantic',
-            '-module', $fipsmodule, '-provider_name', 'fips',
-            '-section_name', 'fips_sect', '-out', $fipsmoduleconf])),
-   "fips install");
+#ok(run(app(['openssl', 'fipsinstall', '-pedantic',
+#            '-module', $fipsmodule, '-provider_name', 'fips',
+#            '-section_name', 'fips_sect', '-out', $fipsmoduleconf])),
+#   "fips install");
diff --git a/test/recipes/01-test_fipsmodule_cnf.t b/test/recipes/01-test_fipsmodule_cnf.t
old mode 100644
new mode 100755
index ce594817d5..4530a46dd0
--- a/test/recipes/01-test_fipsmodule_cnf.t
+++ b/test/recipes/01-test_fipsmodule_cnf.t
@@ -31,7 +31,8 @@ plan tests => 1;
 my $fipsmodule = bldtop_file('providers', platform->dso('fips'));
 my $fipsmoduleconf = bldtop_file('test', 'fipsmodule.cnf');
 
+ok(1, "fipsinstall verification skipped without FIPS provider module");
 # verify the $fipsconf file
-ok(run(app(['openssl', 'fipsinstall',
-            '-in',  $fipsmoduleconf, '-module', $fipsmodule, '-verify'])),
-   "fipsinstall verify");
+#ok(run(app(['openssl', 'fipsinstall',
+#            '-in',  $fipsmoduleconf, '-module', $fipsmodule, '-verify'])),
+#   "fipsinstall verify");
diff --git a/test/recipes/03-test_fipsinstall.t b/test/recipes/03-test_fipsinstall.t
old mode 100644
new mode 100755
index 3dcbe67c6d..1a5a475d91
--- a/test/recipes/03-test_fipsinstall.t
+++ b/test/recipes/03-test_fipsinstall.t
@@ -22,6 +22,6 @@ use lib srctop_dir('Configurations');
 use lib bldtop_dir('.');
 use platform;
 
-plan skip_all => "Test only supported in a fips build" if disabled("fips");
+plan skip_all => "FIPS provider module is not built by the main OpenSSL package";
 
 # Compatible options for pedantic FIPS compliance
-- 
2.51.0
